File: Sorting/bubble_sort.py
from typing import List


def bubble_sort(arr: List[int]):
    n = len(arr)
    for i in range(n - 1):
        swapped = False
        for j in range(n - 1 - i):
            if arr[j] > arr[j + 1]:
                swapped = True
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

        if not swapped:
            break


if __name__ == "__main__":
    arr = [6, 5, 3, 1, 8, 7, 2, 4]
    bubble_sort(arr)
    print(arr)

File: Sorting/BubbleSort.java
import java.util.Arrays;

/**
 * BubbleSort Algorithm
 */
public class BubbleSort {

    public static void main(String[] args) {
        int[] arr = new int[] { 6, 5, 3, 1, 8, 7, 2, 4 };
        bubbleSort(arr);
        System.out.println(Arrays.toString(arr));
    }

    private static void bubbleSort(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }
}
